sonar-kotlin-checks/src/main/java/org/sonarsource/kotlin/checks/UnsuitedFindFunctionWithNullComparisonCheck.kt

```diff
@@ -47,32 +50,30 @@ import org.sonarsource.kotlin.api.reporting.message
 @Rule(key = "S6528")
 class UnsuitedFindFunctionWithNullComparisonCheck : CallAbstractCheck() {
 
-    override val functionsToVisit = listOf(
-        FunMatcher(qualifier = "kotlin.collections") {
-            withNames("find", "findLast", "firstOrNull", "lastOrNull")
-            withArguments("kotlin.Function1")
-        }
-    )
+    override val functionsToVisit = listOf(FunMatcher(qualifier = "kotlin.collections") {
+        withNames("find", "findLast", "firstOrNull", "lastOrNull")
+        withArguments("kotlin.Function1")
+    })
 
     override fun visitFunctionCall(callExpression: KtCallExpression, resolvedCall: ResolvedCall<*>, kotlinFileContext: KotlinFileContext) {
-        callExpression.findClosestAncestorOfType<KtBinaryExpression>()
-            ?.takeIf { it.right!!.isNull() || it.left!!.isNull() }
+        callExpression.findClosestAncestorOfType<KtBinaryExpression>()?.takeIf { it.right!!.isNull() || it.left!!.isNull() }
             ?.let { report(it, callExpression, kotlinFileContext) }
     }
 
     private fun report(nullComparisonExpr: KtBinaryExpression, callExpression: KtCallExpression, kotlinFileContext: KotlinFileContext) {
         // callExpression has argument a lambda expression with a single parameter, due to the functionsToVisit FunMatchers
         val lambda = callExpression.valueArguments[0].getArgumentExpression() as KtLambdaExpression
 
-        val lambdaBinaryExpr = lambda.bodyExpression!!.firstChild as? KtBinaryExpression
+        val lambdaBinaryExpr = lambda.bodyExpression!!.children[0] as? KtBinaryExpression
```

Comment:
this way we can cover:
```
        list.find {// Noncompliant {{Replace with `list.contains(5)`.}}
            it == 5
        } != null
```

Reply:
So `.firstChild` and `.children[0]` behaves differently? Maybe worth a small comment in the code!